proposed/2022/globalpackagereference.md

```diff
@@ -0,0 +1,125 @@
+# Global Package Reference for Central Package Management
+
+- [Jeff Kluge](https://github.com/jeffkl)
+- 2022-05-05
+- [[Feature]: CPVM equivalent of GlobalPackageReference? (#10159)](https://github.com/NuGet/Home/issues/10159)
+- GitHub PR (GitHub PR link)
+
+## Summary
+<!-- One-paragraph description of the proposal. -->
+A common pattern in repositories is to have every project in the tree reference packages that are used for development purposes only.  This includes
+packages that provide assembly versioning, signing, code analysis, etc.  We want to provide a built-in mechanism for users to define packages that
+are used by every project but only for development purposes.
+
+## Motivation 
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Users can manually reference the package in a common import like Directory.Build.props but there is no built-in safe guards to ensure that they don't
+accidentally consume more assets than just development assets.  For example, you don't want to have a package's compile-time assets be referenced by
+every assembly in your repository.  Instead, individual projects should express their build time dependencies and project references should receive
+them transitively.  But you should not need to add a package reference in order to have your whole tree use code analysis or assembly versioning.
+
+## Explanation
+
+### Functional explanation
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+We'll introduce a new MSBuild item group named `<GlobalPackageReference />` which will define packages that every project should reference but has
+default metadata to ensure that only package assets related to development purposes are used.
+
+**NOTE:** This functionality will only be enabled if a user has opt-ed into [Central Package Management (CPM)](https://docs.microsoft.com/nuget/consume-packages/central-package-management).
+
+```xml
+<Project>
+  <ItemGroup>
+    <GlobalPackageReference Include="Nerdbank.GitVersioning" Version="1.0.0" />
```

Comment:
I don't think the name `GlobalPackageReference` is correct. 

The `Global` part is a function of where this is written, not the value of the item itself. 

The itself is what was said above...a development dependency reference. 

So if we were to add this functionality, the name should be changed.

Reply:
Naming suggestion: `CentralPackageReference`, to reflect Central Package Management usage.

Reply:
Naming suggestion: `CentralDevPackageReference`, toe reflect CPM usage on a development dependency.

Reply:
I would suggest `BuildOnlyPackageReference` or `DevOnlyPackageReference` but I suspect `GlobalPackageReference` will be an appropriate naming considering the purpose of it. Also `CentralPackageReference` doesn't sound right compared with the former. I can also suggest `SharedPackageReference` and `CommonPackageReference` but @jeffkl would've already pondered about it when he named it. So, it's better to ask him.

Reply:
I would suggest `BuildOnlyPackageReference` or `DevOnlyPackageReference` but I suspect `GlobalPackageReference` will be an appropriate name considering the location and the **supposed** purpose of it. The default is development only dependencies but it's not stopping us from including other assets.

Also `CentralPackageReference` doesn't sound right compared with the former. I can also suggest `SharedPackageReference` and `CommonPackageReference` but @jeffkl would've already pondered about it when he named it. So, it's up to him.

Reply:
After two months of collecting feedback, `GlobalPackageReference` appears to be the most voted for naming based on internal and external surveys.

https://twitter.com/nuget/status/1545037921455153155

I think we should move forward with `GlobalPackageReference` at this point.